relay-profiling/src/lib.rs

```diff
@@ -133,15 +137,22 @@ fn minimal_profile_from_json(data: &[u8]) -> Result<MinimalProfile, ProfileError
     serde_json::from_slice(data).map_err(ProfileError::InvalidJson)
 }
 
-pub fn expand_profile(payload: &[u8]) -> Result<Vec<u8>, ProfileError> {
-    let profile: MinimalProfile = minimal_profile_from_json(payload)?;
-    match profile.version {
+pub fn expand_profile(payload: &[u8]) -> Result<(EventId, Vec<u8>), ProfileError> {
+    let profile: MinimalProfile = match minimal_profile_from_json(payload) {
+        Ok(profile) => profile,
+        Err(err) => return Err(err),
+    };
+    let processed_payload = match profile.version {
         Version::V1 => parse_sample_profile(payload),
         Version::Unknown => match profile.platform {
             Platform::Android => parse_android_profile(payload),
             Platform::Cocoa => parse_cocoa_profile(payload),
-            _ => Err(ProfileError::PlatformNotSupported),
+            _ => return Err(ProfileError::PlatformNotSupported),
         },
+    };
+    match processed_payload {
+        Ok(payload) => Ok((profile.event_id, payload)),
+        Err(err) => Err(err),
     }
```

Comment:
_nit_: Can be written with `map`:

```suggestion
    processed_payload.map(|payload| (profile.event_id, payload))
```